client/src/client.rs

```diff
@@ -22,9 +23,80 @@ use small::SmallStr;
 
 use crate::{
     config::Configuration,
-    http_client::{self, StatusCode, WebSocketError, WebSocketMessage},
+    http::{Headers as HttpHeaders, Method as HttpMethod, RequestBuilder, Response, StatusCode},
+    http_default::{self, DefaultRequestBuilder, WebSocketError, WebSocketMessage},
 };
 
+/// General trait for all response handlers
+pub trait ResponseHandler<O, T = Vec<u8>> {
+    /// Function to parse HTTP response with body `T` to output `O`
+    ///
+    /// # Errors
+    /// May fail by some reason, depends on implementation
```

Comment:
```suggestion
    /// Implementation dependent. 
```
Which means you should add docs to the implementations too.